### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by usorfaitheloho**
In this PR,I did the following:
## Project requirements

- Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)
Think about how you can use two decorators in order to capitalize and trim people's names.
In the following steps you will get instructions for implementing the Decorator design pattern for this case.
Interface
- Create a class Nameable.
Implement a method called correct_name that will raise a NotImplementedError.
Turn your Person class to Nameable
Make sure that your Person class inherits from Nameable
Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

- Prepare base Decorator
Make sure that it inherits from Nameable.
In the constructor assign a nameable object from params to an instance variable.
Implement the correct_name method that returns the result of the correct_name method of the @nameable.
Prepare CapitalizeDecorator and TrimmerDecorator

For the CapitalizeDecorator:
Create a class that inherits from the base Decorator class.
Implement a method correct_name that capitalizes the output of @nameable.correct_name.

For the TrimmerDecorator:
Create a class that inherits from the base Decorator class.
Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: trimmerdecorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0...10]
```
Comment (Deepakdanger) [inline trimmerdecorator.rb]: - [ ] Here we have to trim the name when the `correct_name` is greater than 10 letters, I  suggest using this here:-
```
@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
```

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class Decorator < Nameable
```
Comment (Deepakdanger) [inline decorator.rb:3]: - [ ] After declaring class , I suggest to use `attr_accessor :nameable` here. Kindly add this.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @usorfaitheloho, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmerdecorator.rb (hunk 1 above)
Comment (usorfaitheloho) [reply trimmerdecorator.rb]: thanks for the review.
it has been updated

File: decorator.rb (hunk 2 above)
Comment (usorfaitheloho) [reply decorator.rb:3]: Thanks for the update.
it has been resolved

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @usorfaitheloho    :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by usorfaitheloho** [comment]
Hello @hiromataba 

Thanks for your review

**Pull request merged**